content/open-source/sigstore/cosign/how-to-verify-file-signatures-with-cosign.md

```diff
@@ -0,0 +1,75 @@
+weight: 005
+toc: true
+---
+
+Cosign can verify more than just containers. Blobs, or binary large objects, and standard files can be verified in a similar way. You can publish a blob or other artifact to an OCI (Open Container Initiative) or elsewhere and then use Cosign to verify its signature. This tutorial assumes you have Cosign installed, which you can achieve by following our [How to Install Cosign guide](../how-to-install-cosign/).
+
+We’ll verify a binary artifact, in this case, a copy of [`apko`](/open-source/apko/overview/), which is a command-line tool for building container images using a declarative language based on YAML. The methods in this tutorial apply to any blob file Cosign has signed with a keyless signature.
+
+### Verifying a binary with Cosign keyless signatures
+
+All `apko` releases are released with [keyless signatures using Cosign](https://edu.chainguard.dev/open-source/sigstore/cosign/an-introduction-to-cosign/#keyless-signing). You can verify the signature for an `apko` release using the `cosign` tool directly, or by calculating the SHA256 hash of the release and finding the corresponding Rekor transparency log entry.
+
+If you would like to learn how to verify a binary using Rekor or curl, follow the steps in our guide [How to Verify File Signatures with Rekor or curl](open-source/sigstore/rekor/how-to-verify-file-signatures-with-rekor-or-curl/).
+
+We'll use the `apko_0.6.0_linux_arm64.tar.gz` tar archive from the apko [GitHub Release v0.6.0 page](https://github.com/chainguard-dev/apko/releases/tag/v0.6.0) in this example.
+
+There are three URLs from the list of assets on that page that you will need to copy:
+
+1. The release itself: `https://github.com/chainguard-dev/apko/releases/download/v0.6.0/apko_0.6.0_linux_amd64.tar.gz`
+2. The signature file: `https://github.com/chainguard-dev/apko/releases/download/v0.6.0/apko_0.6.0_linux_amd64.tar.gz.sig`
+3. The public certificate: `https://github.com/chainguard-dev/apko/releases/download/v0.6.0/apko_0.6.0_linux_amd64.tar.gz.crt`
+
+With these URLs, construct (or copy) the following command to verify the tar archive:
+
+```sh
+COSIGN_EXPERIMENTAL=1 cosign verify-blob \
+   --signature https://github.com/chainguard-dev/apko/releases/download/v0.6.0/apko_0.6.0_linux_amd64.tar.gz.sig \
+   --certificate https://github.com/chainguard-dev/apko/releases/download/v0.6.0/apko_0.6.0_linux_amd64.tar.gz.crt \
+   https://github.com/chainguard-dev/apko/releases/download/v0.6.0/apko_0.6.0_linux_amd64.tar.gz
+```
+
+Running the command may take a moment, but when it completes you will receive the following output:
+
+```
+tlog entry verified with uuid: 9ec23abb6326ebbaa6932f720847080e8f5e0b2925a1643b63962691917c8137 index: 8538988
+Verified OK
+```
+
+If any of the URLs are incorrect, of if there was a problem with the apko release file, a mismatching signature or certificate, or if the release file was not signed, you will receive an error like the following:
```

Comment:
This is a run on, do you want to break it apart into two or more sentences?

Reply:
Hah you caught that too, I knew when I was writing it that there had to be a better way, but get-the-draft-finished-itis won out. Will figure out how to rephrase it